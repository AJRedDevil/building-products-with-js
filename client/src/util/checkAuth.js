const key = 'user.token';

export const isAuthenticated = () => localStorage.getItem(key);
