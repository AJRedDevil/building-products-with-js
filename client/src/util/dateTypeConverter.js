import moment from 'moment';

export const convertStringToDate = strDate => moment(strDate).toDate();
export const convertDateToISOString = dateType => moment(dateType).toISOString();
