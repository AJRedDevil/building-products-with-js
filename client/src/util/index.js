export {PrivateRoute} from './requireAuth';
export {isAuthenticated} from './checkAuth';
export {
  loginErrorToMessage, registerErrorToMessage, ajaxErrorToMessage,
} from './errorToMessage';
export {signRequest} from './signRequest';
export {MyPropType} from './ourPropTypes';
