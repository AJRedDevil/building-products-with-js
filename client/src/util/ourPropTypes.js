import PropTypes from 'prop-types';

export const QuestionPropType = PropTypes.shape({
  answers: PropTypes.array,
  creationDate: PropTypes.string,
  expirationDate: PropTypes.string,
  id: PropTypes.string,
  owner: PropTypes.string,
  text: PropTypes.string,
});

export const NotificationPropType = PropTypes.shape({
  id: PropTypes.number,
  alertType: PropTypes.string,
  text: PropTypes.string,
});

export const NotificationDefaultProp = {
  id: 0,
  alertType: '',
  text: '',
};
