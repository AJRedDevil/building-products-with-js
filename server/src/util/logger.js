import winston from 'winston';

let level;
switch (process.env.NODE_ENV) {
  case 'testing':
    level = 'error';
    break;
  case 'production':
    level = 'info';
    break;
  default:
    level = 'debug';
    break;
}

export const logger = new winston.Logger({
  transports: [
    new winston.transports.Console({
      level,
      colorize: true,
      timestamp: true,
      prettyPrint: true,
      label: 'experts-server',
    }),
  ],
});

// create stream for morgan
logger.stream = {
  write: message => logger.info(message),
};

export {logger as default};
