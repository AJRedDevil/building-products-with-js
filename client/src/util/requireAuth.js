// npm packages
import React from 'react';
import {Redirect, Route} from 'react-router-dom';
import PropTypes from 'prop-types';

// our packages
import {isAuthenticated} from './checkAuth';

export const PrivateRoute = ({component: Component, ...rest}) => (
  <Route
    {...rest}
    render={props => (
      isAuthenticated() ? (
        <Component {...props} />
      ) : (
        <Redirect to={{
          pathname: '/login',
          state: {from: props.location.pathname},
          }}
        />
      )
    )
    }
  />
);
PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired,
  location: PropTypes.shape({pathname: PropTypes.string}),
};
PrivateRoute.defaultProps = {
  location: {pathname: ''},
};
