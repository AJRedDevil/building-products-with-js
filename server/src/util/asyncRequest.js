import {logger} from './logger';

const asyncRequest = handler =>
  (req, res) =>
    handler(req, res).catch(e => logger.error('error diring request', e));

export default asyncRequest;
