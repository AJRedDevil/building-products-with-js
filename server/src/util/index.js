import hash from './hash';
import asyncRequest from './asyncRequest';

export {logger} from './logger';
export {asyncRequest, hash};
