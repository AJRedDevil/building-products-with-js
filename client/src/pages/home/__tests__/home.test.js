// npm packages
import React from 'react';
import configureMockStore from 'redux-mock-store';

// our packages
import HomePage, {Home} from '../index';
import {convertDateToISOString} from '../../../util';

// create mockstore
const mockStore = configureMockStore();

const user = {
  id: 0,
  login: 'test',
  registrationDate: convertDateToISOString(new Date(2017, 1, 1, 1, 1, 1)),
};

test('# Home page wrapper', () => {
  const store = mockStore({auth: {user}, questions: []});
  const wrapper = shallow(<HomePage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Home page', () => {
  const questions = [];
  const fetchQuestions = () => questions.push({
    id: 0,
    owner: user,
    text: 'Question text',
    answers: [{answer: 'Test answer'}],
  });

  const component = (
    <Home
      user={user}
      questions={questions}
      fetchQuestions={fetchQuestions}
      doAnswer={() => {}}
    />
  );
  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
