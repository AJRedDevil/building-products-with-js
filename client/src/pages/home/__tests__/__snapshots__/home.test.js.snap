// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Home page 1`] = `
<div>
  <Navbar
    current="/"
    user={
      Object {
        "id": 0,
        "login": "test",
        "registrationDate": "2017-01-31T19:16:01.000Z",
      }
    }
  />
  <div>
    <Connect(Question)
      key="0"
      onAnswer={[Function]}
      question={
        Object {
          "answers": Array [
            Object {
              "answer": "Test answer",
            },
          ],
          "id": 0,
          "owner": Object {
            "id": 0,
            "login": "test",
            "registrationDate": "2017-01-31T19:16:01.000Z",
          },
          "text": "Question text",
        }
      }
    />
  </div>
</div>
`;

exports[`# Home page wrapper 1`] = `
<Home
  doAnswer={[Function]}
  fetchQuestions={[Function]}
  questions={
    Array [
      Object {
        "answers": Array [],
        "creationDate": "",
        "expirationDate": "",
        "id": "",
        "owner": Object {
          "id": "",
          "login": "",
          "registrationDate": "",
        },
        "text": "",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "id": 0,
      "login": "test",
      "registrationDate": "2017-01-31T19:16:01.000Z",
    }
  }
/>
`;
